src/network/ClientConnectionManager.ts

```diff
@@ -277,8 +277,15 @@ export class ClientConnectionManager extends EventEmitter {
         return timedPromise(
             connectionResolver.promise,
             this.connectionTimeoutMillis,
-            new HazelcastError(`Connection timed out to address ${address}.`)
-        ).finally(() => this.pendingConnections.delete(addressKey));
+            new HazelcastError('Connection timed out to address ' + address.toString())
```

Comment:
Since the connection timeout logic is moved inside the `connectXXXSocket` functions, why do we still need a `timedPromise` here?

Reply:
That's because we still have other method invocations here where we do network comms without preserving the timeout. Examples of such methods are `translateAddress` (which may lead to a network call to the Hazelcast Cloud server) and `authenticateOnCluster`. That's why I decided to keep the per-`getOrConnect` timeout in place.

Reply:
WDYT?

Reply:
I see, but wouldn't this be a problem for the authentication part. In there, we have 120 seconds of authentication timeout (same as the Java client), and because of this check, it might timeout earlier than that

Reply:
Yes, keeping things consistent with Java client certainly makes sense. Removed the `timedPromise` part in 15279db69c82dc23ce97e5a4f0aab289e2727555